devlib/target.py

```diff
@@ -153,6 +155,7 @@ def __init__(self,
                  ):
         self.connection_settings = connection_settings or {}
         self.platform = platform or Platform()
+        self.connection_settings['platform'] = self.platform
```

Comment:
Yes, but isn't that the same for other arguments as well? Like for example, device in AdbConnection? (Sorry, not trying to be annoying, just want to learn)